src/librustc/middle/mem_categorization.rs

```diff
@@ -333,8 +333,10 @@ impl MutabilityCategory {
     fn from_local(tcx: TyCtxt, id: ast::NodeId) -> MutabilityCategory {
         let ret = match tcx.hir.get(id) {
             hir_map::NodeLocal(p) => match p.node {
-                PatKind::Binding(bind_mode, ..) => {
-                    if bind_mode == hir::BindByValue(hir::MutMutable) {
+                PatKind::Binding(bm, ..) => {
+                    // FIXME(tschottdorf): no tables in scope.
```

Comment:
just add an argument to the fn; it's local to this file, and the callers can use `self.tables`